### PR TITLE
status: using dynamic api instead of kubectl

### PR DESCRIPTION
**Pull request by subhamkrai**
**Description of your changes:**
to get crd resources let's start using dynamic api
of kubernetes instead of using `kubectl` cli directly.
It make it platform independent wether it is `kubectl`
kubernetes or `oc` openshift.

Note: it changes how the command output used to be.

Signed-off-by: subhamkrai <srai@redhat.com>

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] CI tests has been updated, if necessary.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.

### CODE REVIEW

File: pkg/rook/status.go [hunk 1]
```diff
@@ -17,53 +17,51 @@ limitations under the License.
...
-func concatenateJsonflag(arg string) string {
-	return arg + " -o json"
+func printStatus(ctx context.Context, k8sclientset *k8sutil.Clientsets, clusterNamespace, resource string) {
+	items, err := k8sclientset.ListResourcesDynamically(ctx, crds.CephRookIoGroup, crds.CephRookResourcesVersion, resource, clusterNamespace)
+	if err != nil {
+		logging.Fatal(err)
+	}
+
+	if len(items) == 0 {
+		logging.Warning("resource %s was not found on the cluster", resource)
```
Comment (travisn) [inline pkg/rook/status.go:51]: A warning implies something might be wrong, but this seems normal.
```suggestion
		logging.Info("resource %s was not found on the cluster", resource)
```

**Review by travisn — CHANGES_REQUESTED** [review]
For the commit prefix, I'd suggest just single words like `status`.

File: pkg/rook/status.go (hunk 1 above)
Comment (subhamkrai) [reply pkg/rook/status.go]: done

File: docs/rook.md [hunk 2]
```diff
@@ -46,173 +45,193 @@ kubectl rook-ceph rook version
...
+#   deviceClasses:
+#   - name: hdd
+#   osd:
+#     storeType:
+#       bluestore: 1
+# version:
+#   image: quay.io/ceph/ceph:v18
+#   version: 18.2.1-0
+
+# Warning: resource cephblockpoolradosnamespaces was not found on the cluster
```
Comment (travisn) [inline docs/rook.md:156]: These warnings are all Info now, need to update the example

**Review by travisn — CHANGES_REQUESTED** [review]

File: docs/rook.md (hunk 2 above)
Comment (subhamkrai) [reply docs/rook.md]: missed, updating this earlier

**Review by travisn — APPROVED** [review]

**Pull request merged**